### PR TITLE
Input validation on the project creation form

### PR DESCRIPTION
**Pull request by jfucci**
githubUsername and githubProjectName having slashes in them was causing issues. This code makes it so that they cannot have a forward slash.

should fix issue #617

### CODE REVIEW

**Comment by jfucci** [comment]
should fix issue #617

**Comment by jfucci** [comment]
newest commit fixes #603... which also has to do with input validation

**Comment by jfucci** [comment]
latest 2 commits fix issue #604

**Comment by kmcnellis** [comment]
Once the field is verified with a / character, the warning never goes away & the user cannot submit
screen shot 2017-04-11 at 1 37 04 pm

**Comment by kmcnellis** [comment]
The various inputs with verification should have live error-checking using 
`ng-model-options="{ updateOn: 'default blur', debounce: { 'default': 500, 'blur': 0 } }" `
as seen in https://github.com/rcos/Observatory3/blob/master/client/app/account/settings/settings.html

File: client/components/editProject/editProject.html [hunk 1]
```diff
@@ -27,12 +27,18 @@ <h3 ng-hide="editing" class="modal-title">New Project</h3>
...
+        <p class="help-block" ng-show="editProjectForm.githubUsername.$error.mongoose && submitted">
+          This field cannot have a '/' character
+        </p>
       </div>
       <div class="form-group" ng-class="{ 'has-success': editProjectForm.githubProjectName.$valid && submitted,
       'has-error': editProjectForm.githubProjectName.$invalid && submitted }">
         <p class="help-block" ng-show="editProjectForm.githubProjectName.$error.required && submitted">
           The project's github name is required
         </p>
+        <p class="help-block" ng-show="editProjectForm.githubProjectName.$error.mongoose && submitted">
```
Comment (kmcnellis) [inline client/components/editProject/editProject.html]: These two `This field cannot have a '/' character` messages shouldn't stack. Lets just use 
`ng-show="(editProjectForm.githubUsername.$error.mongoose || editProjectForm.githubProjectName.$error.mongoose) && submitted"` or something of the sort

File: package.json [hunk 2]
```diff
@@ -80,7 +80,6 @@
     "grunt-newer": "^1.1.1",
     "grunt-ng-annotate": "^1.0.1",
     "grunt-ng-constant": "^2.0.1",
-    "grunt-node-inspector": "^0.4.1",
```
Comment (kmcnellis) [inline package.json]: This shouldn't really be in this PR, but ok. 🤷‍♂️

Comment (jfucci) [inline package.json]: It was the first thing I had to do to get grunt serve to run so I just committed it. Can revert if necessary

**Comment by jfucci** [comment]
I cannot reproduce the issue with not being able to re-send the form after sending one with slashes. Are you sure the branch is fully up to date? I had that issue at some point, but not now:
http://imgur.com/a/7N7YZ

File: client/components/editProject/editProject.html (hunk 1 above)
Comment (jfucci) [inline client/components/editProject/editProject.html]: latest commit fixes this

File: package.json (hunk 2 above)
Comment (kmcnellis) [inline package.json]: Yeah, it doesn't work with the newest version of node. You can leave it in

**Comment by kmcnellis** [comment]
Looks good now, thanks.

**Pull request merged**